app/models/stripe_record.rb

```diff
@@ -182,17 +182,32 @@ def self.amount_to_ruby(amount_stripe_denomination, iso_currency)
     amount_stripe_denomination.to_i
   end
 
-  def self.create_from_api(api_record, parameters, account_id, parent_record = nil)
-    StripeRecord.create!(
-      stripe_record_type: api_record.object,
-      parameters: parameters,
-      stripe_id: api_record.id,
-      amount_stripe_denomination: api_record.amount,
-      currency_code: api_record.currency,
-      stripe_status: api_record.status,
-      account_id: account_id,
-      parent_record: parent_record,
-    )
+  def self.create_or_update_from_api(api_record, parameters = nil, account_id = nil, parent_record = nil)
+    existing_record = StripeRecord.find_by(stripe_id: api_record.id)
+    new_record = nil
+
+    if existing_record.present?
+      existing_record.update!(
+        stripe_record_type: api_record.object,
+        stripe_id: api_record.id,
+        amount_stripe_denomination: api_record.amount,
+        currency_code: api_record.currency,
+        stripe_status: api_record.status,
+      )
+    else
+      new_record = StripeRecord.create!(
+        stripe_record_type: api_record.object,
+        parameters: parameters,
+        stripe_id: api_record.id,
+        amount_stripe_denomination: api_record.amount,
+        currency_code: api_record.currency,
+        stripe_status: api_record.status,
+        account_id: account_id,
+        parent_record: parent_record,
+      )
+    end
+
+    new_record || existing_record
```

Comment:
This still feels a bit too "manual". I think a more idiomatic way would be to leverage the `create_or_initialize_by` helper, something along the lines of:
```suggestion
  def self.create_or_update_from_api(api_record, parameters = nil, account_id = nil, parent_record = nil)
    record = StripeRecord.find_or_initialize_by(stripe_id: api_record.id) do |new_record|
      new_record.parameters = parameters
      new_record.account_id = account_id
      new_record.parent_record = parent_record
    end.tap do |record|
      record.update!(
        stripe_record_type: api_record.object,
        stripe_id: api_record.id,
        amount_stripe_denomination: api_record.amount,
        currency_code: api_record.currency,
        stripe_status: api_record.status,
      )
    end
```

Reply:
Note that I wrote this freestyle directly in the GH comment panel, you should definitely try whether it works as intended.

Per the documentation of `find_or_initialize_by`, it takes one block as optional argument, which is only executed in the "initialize" case.
The other `tap` could be replaced by
```ruby
record = find_or_initialize_by ...

record.update!(...)

record # as explicit return value
```
but I think since it's only one single `update!` operation on the very variable that we want as return value anyways, tapping is warranted and readable.

Reply:
Altered as suggested, ran the relevant tests locally and it all looks good, we'll see what the CI says.

Also notable that I removed `stripe_record_type: api_record.object,` from the `update` portion - as in my mind, we should only be touching that parameter upon create